include/tvm/runtime/vm/bytecode.h

```diff
@@ -201,6 +203,10 @@ struct Instruction {
       Index alignment;
       /*! \brief The hint of the dtype. */
       DLDataType dtype_hint;
+      /*! \brief The number of dimensions. */
+      uint32_t ndim;
+      /*! \brief The shape of tensor. */
+      int64_t* shape;
```

Comment:
With these information, does the existing field, `allocation_size`, become redundant? How about putting them into a union to save one int64?

Reply:
Done, put `shape` and `allocation_size` into a union.